src/odemis/acq/move.py

```diff
@@ -285,8 +294,18 @@ def _doCryoSwitchSamplePosition(future, stage, focus, align, target):
         else:
             raise ValueError("Unknown target value %s." % target)
 
-        for component, sub_move in sub_moves:
-            run_sub_move(future, component, sub_move)
+        try:
+            for component, sub_move in sub_moves:
+                run_sub_move(future, component, sub_move)
+        except SA_MCError as sa_mcerror:
+            # In case the required movement is invalid/unreachable with the smaract 5dof stage
+            # Move all linear axes first then rotational ones using the fallback_submoves
+            if sa_mcerror.errno != MC_5DOF_DLL.SA_MC_ERROR_POSE_UNREACHABLE:
+                raise
```

Comment:
Let's really avoid having the function depend on some specific driver. I'm mostly not so happy with the imports that it causes. I'd suggest one of these ways:
* Just detect an Exception, and always retry with the fallback moves. In case this doesn't help, it'd probably fail again the same way.
* Let the 5DoF driver catch this specific error and raise a IndexError. Here, catch the IndexError.
* Create a new type of exception like "UnreachablePositionError", which you'd add to model/_components.py. Then you can let the 5DoF driver catch the specific error and raise UnreachablePositionError.

I'd recommend the 2nd option, but it's up to you.